src/ray/protobuf/experimental/autoscaler.proto

```diff
@@ -271,6 +275,33 @@ message GetClusterStatusReply {
   ClusterResourceState cluster_resource_state = 2;
 }
 
+enum DrainNodeReason {
+  // The node has been idle for long enough time.
+  // Drain request with this reason is rejectable.
+  IDLE_TERMINATION = 0;
+  // The node will be preempted.
+  // Drain request with this reason is not rejectable.
+  PREEMPTION = 1;
+}
+
+message DrainNodeRequest {
+  // The id of the node to be drained.
+  bytes node_id = 1;
+  // The reason why the node will be drained.
+  // This information will be used to
+  // decide whether the request is rejectable.
+  DrainNodeReason reason = 2;
+  // The detailed drain reason message.
```

Comment:
does it make sense to have a is_rejectable at this request?

Reply:
You mean to replace `reason` or use together with `reason`?

Reply:
A single `is_rejectable` is probably not enough since different reason can have different reject condition.